src/accelerate/tracking.py

```diff
@@ -151,7 +154,13 @@ def store_init_configuration(self, values: dict):
         """
         self.writer.add_hparams(values, metric_dict={})
         self.writer.flush()
-        logger.info("Stored initial configuration hyperparameters to TensorBoard")
+        project_run_name = time.time()
+        with open(os.path.join(self.logging_dir, project_run_name,  "hparams.yml"), "w") as outfile:
+            try:
+                yaml.dump(values, outfile)
+            except yaml.representer.RepresenterError:
+                logger.info("Serialization to store hyperparmeters failed")
```

Comment:
We should change this to actually raise the error, otherwise this will silently fail.

```suggestion
            except yaml.representer.RepresenterError:
                logger.error("Serialization to store hyperparameters failed")
                raise
```
I think just doing `logger.error` here is fine enough

Reply:
Yes, I have made the changes.